File: backend/src/main/java/com/alfredosoto/portfolio/service/CommentService.java
package com.alfredosoto.portfolio.service;

import com.alfredosoto.portfolio.entity.CommentEntity;
import java.util.List;

public interface CommentService {
    CommentEntity addComment(CommentEntity comment);
    List<CommentEntity> getApprovedComments();
    List<CommentEntity> getAllComments(); // Admin
    CommentEntity approveComment(String id);
    CommentEntity updateComment(String id, String content, String token);
    void deleteComment(String id, String token);
}

File: frontend/src/app/services/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Comment } from '../models/comment.model';

@Injectable({
  providedIn: 'root'
})
export class CommentService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/comments`;

  getApprovedComments(): Observable<Comment[]> {
    return this.http.get<Comment[]>(this.apiUrl);
  }

  addComment(comment: Comment): Observable<Comment> {
    return this.http.post<Comment>(this.apiUrl, comment);
  }

  getAllComments(): Observable<Comment[]> {
    return this.http.get<Comment[]>(`${this.apiUrl}/admin`);
  }

  approveComment(id: string): Observable<Comment> {
    return this.http.put<Comment>(`${this.apiUrl}/${id}/approve`, {});
  }

  updateComment(id: string, content: string, token: string): Observable<Comment> {
    return this.http.put<Comment>(`${this.apiUrl}/${id}?token=${token}`, { content });
  }

  deleteComment(id: string, token?: string): Observable<void> {
    const url = token ? `${this.apiUrl}/${id}?token=${token}` : `${this.apiUrl}/${id}`;
    return this.http.delete<void>(url);
  }
}
